x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix.tsx

```diff
@@ -0,0 +1,313 @@
+
+    const view = new View(parse(vgSpec))
+      .logLevel(Warn)
+      .renderer('canvas')
+      .tooltip(new Handler().call)
+      .initialize(`#${htmlId}`);
+
+    view.runAsync(); // evaluate and render the view
+  }, [resultsField, splom, color, legendType, dynamicSize]);
+
+  return (
+    <>
+      {splom === undefined && (
+        <EuiText textAlign="center">
+          <EuiSpacer size="l" />
+          <EuiLoadingSpinner size="l" />
+          <EuiSpacer size="l" />
+        </EuiText>
+      )}
+      {splom !== undefined && (
+        <>
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.splom.fieldSelectionLabel', {
+                  defaultMessage: 'Fields',
+                })}
+                display="rowCompressed"
+                fullWidth
+              >
+                <EuiComboBox
+                  compressed
+                  fullWidth
+                  placeholder={i18n.translate('xpack.ml.splom.fieldSelectionPlaceholder', {
+                    defaultMessage: 'Select fields',
+                  })}
+                  options={fieldOptions}
+                  selectedOptions={fields.map((d) => ({
+                    label: d,
+                  }))}
+                  onChange={fieldsOnChange}
+                  isClearable={true}
+                  data-test-subj="mlScatterplotMatrixFieldsComboBox"
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            <EuiFlexItem style={{ width: '200px' }} grow={false}>
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.splom.SampleSizeLabel', {
+                  defaultMessage: 'Sample size',
+                })}
+                display="rowCompressed"
+                fullWidth
+              >
+                <EuiSelect
+                  compressed
+                  options={sampleSizeOptions}
+                  value={fetchSize}
+                  onChange={(e) => fetchSizeOnChange(e)}
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            <EuiFlexItem style={{ width: '120px' }} grow={false}>
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.splom.RandomScoringLabel', {
+                  defaultMessage: 'Random scoring',
+                })}
+                display="rowCompressed"
+                fullWidth
+              >
+                <EuiSwitch
+                  name="mlScatterplotMatrixRandomizeQuery"
+                  label={randomizeQuery ? TOGGLE_ON : TOGGLE_OFF}
+                  checked={randomizeQuery}
+                  onChange={() => randomizeQueryOnChange()}
+                  disabled={isLoading}
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            {resultsField !== undefined && legendType === undefined && (
+              <EuiFlexItem style={{ width: '120px' }} grow={false}>
```

Comment:
not sure if a fixed width works well for responsive layout

Reply:
Since not all items have a fixed width it works, the `Fields` item will shrink.